### PR TITLE
Adding new loss params: lambda jitter

### PR DESCRIPTION
**Pull request by TonyCongqianWang**
Adds perturbations to the lambda parameter used for loss calculation.

Perturbation amount is controlled by newly added parameters.

This allows recipes to work without a lambda schedule (i.e. with lambda_start = lambda_end).

Despite the name, this is the "secret sauce" behind the success of the recipe https://github.com/vondele/nettest/pull/329

### CODE REVIEW

File: model/lightning_module.py [hunk 1]
```diff
@@ -190,6 +195,17 @@ def on_test_epoch_end(self):
 
     def on_save_checkpoint(self, checkpoint):
         self.optimizer_wrapper.on_save_checkpoint(self, checkpoint)
+        checkpoint["jitter_buffer_value"] = self.jitter_buffer
```
Comment (xu-shawn) [inline model/lightning_module.py]: why do we set persistent=false but then explicitly code saving the variable?

Comment (TonyCongqianWang) [reply model/lightning_module.py]: this way it is not saved in the state dict, but in the top level ckpt dictionary instead. This means that it will be continued when using `resume_from_checkpoint` but reset when using `resume_from_model`.... somewhat complicated I know, but I am not aware of any better solution

Comment (xu-shawn) [reply model/lightning_module.py]: and what is the reason we need to have those different behaviors?

Comment (TonyCongqianWang) [reply model/lightning_module.py]: Jitter is not part of the model but part of the training

Comment (xu-shawn) [reply model/lightning_module.py]: hmm I guess maybe the best way then is to factor the lambda schedules into a callback (in a future PR)?

Comment (TonyCongqianWang) [reply model/lightning_module.py]: A lightning callback would probably be quite inefficient. But refactoring everything into it's own component should be quite trivial

File: model/lightning_module.py [hunk 2]
```diff
@@ -257,6 +273,22 @@ def step_(self, batch: tuple[Tensor, ...], batch_idx, loss_type):
...
+            jitter_lambda_batch = loss_params.jitter_lambda_batch * math.sqrt(1 - loss_params.jitter_decay_lambda_batch ** 2)
+            batch_jitter_delta = jitter_lambda_batch * torch.randn_like(self.jitter_buffer)
+            self.jitter_buffer.mul_(loss_params.jitter_decay_lambda_batch).add_(batch_jitter_delta)
+            batch_jitter = self.jitter_buffer.expand_as(scorenet)
+            sample_jitter = torch.randn_like(scorenet) * loss_params.jitter_lambda_sample
+        else:
+            # During evaluation we move allocate all jitter to the sample level for better consistency.
+            batch_jitter = torch.zeros_like(scorenet)
+            sample_jitter = torch.randn_like(scorenet)
+            sample_jitter = sample_jitter * (loss_params.jitter_lambda_sample + loss_params.jitter_lambda_batch)
```
[the remark this answers is not in the document]
Comment (TonyCongqianWang) [reply model/lightning_module.py]: This is desired. The jitter doesn't just create perturbations for the gradient, it does fundamentally change the optimization criterium due to pow-exp > 2 and qp asym.

With a large enough batch size, the gradient is rather stable when only using per sample jitter

File: model/lightning_module.py [hunk 3]
```diff
@@ -190,6 +195,17 @@ def on_test_epoch_end(self):
...
+
+    def on_load_checkpoint(self, checkpoint):
+        trainer = self.__dict__.get("_trainer", None)
+        is_resuming = (
+            trainer is not None and
+            getattr(trainer, "ckpt_path", None) is not None
+        )
+        if is_resuming:
+            if "jitter_buffer_value" in checkpoint:
+                self.jitter_buffer.copy_(checkpoint["jitter_buffer_value"])
```
[the remark this answers is not in the document]
Comment (TonyCongqianWang) [reply model/lightning_module.py]: Half of the suggestion is good, the other half is trash

File: model/lightning_module.py (hunk 1 above)
Comment (TonyCongqianWang) [reply model/lightning_module.py]: Added it now 👍

**Comment by TonyCongqianWang** [comment]
Maybe rerun copilot just in case? I think this is fairly ready to merge now. Recipe ablations also showed that this component is indeed necessary

**Comment by TonyCongqianWang** [comment]
Good finds.. resolved
